python/src/typed/mod.rs

```diff
@@ -60,63 +64,88 @@ fn type_info_for_member(
+            Field::new(
+                &m.name,
+                type_info_for_nestable_type(&sequence.value_type, package_name, messages)?,
+                true,
+            ),
+            true,
+        )
+        .data_type()
+        .clone(),
+    })
+}
+
+fn type_info_for_nestable_type(
+    t: &NestableType,
+    package_name: &str,
+    messages: &HashMap<String, HashMap<String, Message>>,
+) -> Result<DataType> {
     let empty = HashMap::new();
     let package_messages = messages.get(package_name).unwrap_or(&empty);
-    Ok(match &m.r#type {
-        MemberType::NestableType(t) => match t {
-            NestableType::BasicType(t) => match t {
-                BasicType::I8 => todo!(),
-                BasicType::I16 => todo!(),
-                BasicType::I32 => DataType::Int32,
-                BasicType::I64 => todo!(),
-                BasicType::U8 => todo!(),
-                BasicType::U16 => todo!(),
-                BasicType::U32 => todo!(),
-                BasicType::U64 => todo!(),
-                BasicType::F32 => DataType::Float32,
-                BasicType::F64 => DataType::Float64,
-                BasicType::Bool => todo!(),
-                BasicType::Char => todo!(),
-                BasicType::Byte => todo!(),
-            },
-            NestableType::NamedType(name) => {
-                let referenced_message = package_messages
-                    .get(&name.0)
-                    .context("unknown referenced message")?;
-                for_message(messages, package_name, &referenced_message.name)?.fields
-            }
-            NestableType::NamespacedType(t) => for_message(messages, &t.package, &t.name)?.fields,
-            NestableType::GenericString(_) => DataType::Utf8,
-        },
-        MemberType::Array(array) => match array.value_type {
-            _ => todo!(),
-        },
-        MemberType::Sequence(sequence) => match &sequence.value_type {
-            NestableType::NamedType(name) => {
-                let referenced_message = package_messages
-                    .get(&name.0)
-                    .context("unknown referenced message")?;
-                for_message(messages, package_name, &referenced_message.name)?.fields
-            }
-            _ => todo!(),
+    let data_type = match t {
+        NestableType::BasicType(t) => match t {
+            BasicType::I8 => DataType::Int8,
+            BasicType::I16 => DataType::Int16,
+            BasicType::I32 => DataType::Int32,
+            BasicType::I64 => DataType::Int64,
+            BasicType::U8 => DataType::UInt8,
+            BasicType::U16 => DataType::UInt16,
+            BasicType::U32 => DataType::UInt32,
+            BasicType::U64 => DataType::UInt64,
+            BasicType::F32 => DataType::Float32,
+            BasicType::F64 => DataType::Float64,
+            BasicType::Bool => DataType::Boolean,
+            BasicType::Char => DataType::Utf8,
+            BasicType::Byte => DataType::UInt8,
```

Comment:
Is `type_info_for_member` and `type_info_for_nestable_type` even used anymore?

Reply:
It's not, thanks!